src/main/kotlin/no/nav/familie/ef/sak/iverksett/oppgaveforbarn/BarnFyllerÅrOppfølgingsoppgaveService.kt

```diff
@@ -0,0 +1,142 @@
+                gjeldendeBarnRepository.finnBarnTilMigrerteBehandlinger(StønadType.OVERGANGSSTØNAD, dagensDato)
+
+        logger.info("Antall barn i gjeldende behandlinger: ${alleBarnIGjeldendeBehandlinger.size}")
+
+        val skalOpprettes = filtrerBarnSomHarFyltÅr(alleBarnIGjeldendeBehandlinger)
+
+        if (!dryRun) {
+            opprettOppgaveForBarn(skalOpprettes)
+        } else {
+            logger.info("Ville opprettet oppgave for ${skalOpprettes.size} barn.")
+            skalOpprettes.forEach {
+                secureLogger.info(
+                    "Ville opprettet oppgave for barn med fødselsnummer: " +
+                        "${it.fødselsnummer} med alder ${it.alder}"
+                )
+            }
+        }
+    }
+
+    private fun filtrerBarnSomHarFyltÅr(barnTilUtplukkForOppgave: List<BarnTilUtplukkForOppgave>): List<OpprettOppgaveForBarn> {
+        val opprettedeOppgaver = oppgaveRepository.findByTypeAndAlderIsNotNull(Oppgavetype.InnhentDokumentasjon)
+        val skalOpprettes = mutableListOf<OpprettOppgaveForBarn>()
+
+        barnTilUtplukkForOppgave.forEach { barn ->
+            val barnetsAlder = Alder.fromFødselsdato(fødselsdato(barn))
+            if (barnetsAlder != null && barn.fødselsnummerBarn != null && opprettedeOppgaver.none { it.barnPersonIdent == barn.fødselsnummerBarn && it.alder == barnetsAlder }) {
+                skalOpprettes.add(
+                    OpprettOppgaveForBarn(
+                        barn.fødselsnummerBarn,
+                        barn.fødselsnummerSøker,
+                        barnetsAlder,
+                        barn.behandlingId
+                    )
+                )
+            }
+        }
+
+        logger.info("barn til utplukk for oppgave: ${skalOpprettes.size}")
+        return skalOpprettes
+    }
+
+    private fun opprettOppgaveForBarn(opprettOppgaverForBarn: List<OpprettOppgaveForBarn>) {
+        if (opprettOppgaverForBarn.isEmpty()) return
+        val gjeldendeBarnList =
+            gjeldendeBarnRepository.finnEksternFagsakIdForBehandlingId(opprettOppgaverForBarn.map { it.behandlingId }).toSet()
+        gjeldendeBarnList.forEach { gjeldendeBarn ->
+            val opprettOppgaveForEksternId =
+                opprettOppgaverForBarn.firstOrNull { it.fødselsnummer == gjeldendeBarn.barnPersonIdent }
+            val finnesOppgave = oppgaveRepository.findByBehandlingIdAndBarnPersonIdentAndAlder(
+                gjeldendeBarn.behandlingId,
+                gjeldendeBarn.barnPersonIdent,
+                opprettOppgaveForEksternId?.alder
+            ) != null
+            if (!finnesOppgave && opprettOppgaveForEksternId != null) {
+                val opprettOppgaveRequest = lagOppgaveRequestForOppfølgingAvBarnFyltÅr(opprettOppgaveForEksternId, gjeldendeBarn)
+                val opprettetOppgaveId = oppgaveClient.opprettOppgave(opprettOppgaveRequest)
```

Comment:
Husk at dette gjøres i en transaksjon og kan ha flere barn som blir opprettet, hvis det feiler på siste kallet rulles allt tilbake i databasen (tasken) og rekjører og oppretter oppgavene på nytt.
Idag når vi har dette oppsplittet så håndterer iverksett å opprette en task for hver oppgave som sendes i vei. 

Burde kanskje lage en ny task for selve oppgaven som skal opprettes?